packages/assets-controllers/src/CurrencyRateController.test.ts

```diff
@@ -225,64 +274,104 @@ describe('CurrencyRateController', () => {
       messenger,
     });
 
-    expect(controller.state.conversionRate).toBe(0);
-
-    await controller.start();
-    await controller.setNativeCurrency('DAI');
-
-    expect(controller.state.conversionRate).toBe(1);
-
-    await controller.setNativeCurrency(TESTNET_TICKER_SYMBOLS.GOERLI);
+    expect(controller.state.currencyRates).toStrictEqual({
+      ETH: {
+        conversionDate: 0,
+        conversionRate: 0,
+        usdConversionRate: null,
+      },
+    });
 
-    expect(controller.state.conversionRate).toBe(10);
+    await controller.updateExchangeRate('SepoliaETH');
+
+    expect(controller.state.currencyRates).toStrictEqual({
+      ETH: {
+        conversionDate: 0,
+        conversionRate: 0,
+        usdConversionRate: null,
+      },
+      SepoliaETH: {
+        conversionDate: getStubbedDate() / 1000,
+        conversionRate: 10,
+        usdConversionRate: 110,
+      },
+    });
 
     controller.destroy();
   });
 
-  it('should update current currency', async () => {
+  it('should update current currency then clear and refetch rates', async () => {
+    jest.spyOn(global.Date, 'now').mockImplementation(() => getStubbedDate());
     const fetchExchangeRateStub = jest
       .fn()
-      .mockResolvedValue({ conversionRate: 10 });
+      .mockResolvedValue({ conversionRate: 10, usdConversionRate: 11 });
     const messenger = getRestrictedMessenger();
     const controller = new CurrencyRateController({
       interval: 10,
       fetchExchangeRate: fetchExchangeRateStub,
       messenger,
+      state: {
+        currencyRates: {
+          ETH: {
+            conversionDate: 123,
+            conversionRate: 123,
+            usdConversionRate: 123,
+          },
+          BTC: {
+            conversionDate: 100,
+            conversionRate: 200,
+            usdConversionRate: 300,
+          },
+        },
+      },
     });
 
-    expect(controller.state.currentCurrency).toBe('usd');
-
-    await controller.start();
-
-    expect(controller.state.currentCurrency).toBe('usd');
+    expect(controller.state).toStrictEqual({
```

Comment:
[nit] probably don't need this since we just set the state and haven't done anything yet

Reply:
yeah good point

done https://github.com/MetaMask/core/pull/1805/commits/0f67a4e479c77e071d7eb64e191b697fadee1e09